src/runtime_src/core/edge/user/aie/graph_object.cpp

```diff
@@ -17,10 +17,15 @@ namespace zynqaie {
       auto device{xrt_core::get_userpf_device(m_shim)};
       auto drv = ZYNQ::shim::handleCheck(device->get_device_handle());
 
-      if (!drv->isAieRegistered())
-        throw xrt_core::error(-EINVAL, "No AIE presented");
-
-      aieArray = drv->get_aie_array_shared();
+#ifdef XRT_ENABLE_AIE       
+      if (nullptr != m_hwctx) {
+        aieArray = m_hwctx->get_aie_array_shared(); 
+      }
+      else {
+        if (drv->isAieRegistered())
+          aieArray = drv->get_aie_array_shared();
+      }
```

Comment:
```suggestion
      if (m_hwctx)
        aieArray = m_hwctx->get_aie_array_shared(); 
      else if (drv->isAieRegistered())
        aieArray = drv->get_aie_array_shared();
```

And just to confirm, it is okay for `m_hwctx` to be nullptr in line 30?

Reply:
it is still ok, as null check is taken care in called function